modin/engines/ray/utils.py

```diff
@@ -25,3 +29,102 @@ def handle_ray_task_error(e):
+# This prevents a race condition between two threads deserializing functions
+# and trying to import pandas at the same time.
+def _import_pandas(*args):
+    import pandas  # noqa F401
+
+
+def initialize_ray():
+    """Initializes ray based on environment variables and internal defaults."""
+    import ray
+
+    if threading.current_thread().name == "MainThread":
+        import secrets
+
+        plasma_directory = None
+        num_cpus = os.environ.get("MODIN_CPUS", None) or multiprocessing.cpu_count()
+        cluster = os.environ.get("MODIN_RAY_CLUSTER", None)
+        redis_address = os.environ.get("MODIN_REDIS_ADDRESS", None)
+        redis_password = secrets.token_hex(16)
+
+        if cluster == "True":
+            # We only start ray in a cluster setting for the head node.
+            ray.init(
+                address=redis_address or "auto",
+                include_webui=False,
+                ignore_reinit_error=True,
+                redis_password=redis_password,
+                logging_level=100,
+            )
+        elif cluster is None:
+            object_store_memory = os.environ.get("MODIN_MEMORY", None)
+            if os.environ.get("MODIN_OUT_OF_CORE", "False").title() == "True":
+                from tempfile import gettempdir
+
+                plasma_directory = gettempdir()
+                # We may have already set the memory from the environment variable, we don't
+                # want to overwrite that value if we have.
+                if object_store_memory is None:
+                    # Round down to the nearest Gigabyte.
+                    mem_bytes = ray.utils.get_system_memory() // 10 ** 9 * 10 ** 9
+                    # Default to 8x memory for out of core
+                    object_store_memory = 8 * mem_bytes
+            # In case anything failed above, we can still improve the memory for Modin.
+            if object_store_memory is None:
+                # Round down to the nearest Gigabyte.
+                object_store_memory = int(
+                    0.6 * ray.utils.get_system_memory() // 10 ** 9 * 10 ** 9
+                )
+                # If the memory pool is smaller than 2GB, just use the default in ray.
+                if object_store_memory == 0:
+                    object_store_memory = None
+            else:
+                object_store_memory = int(object_store_memory)
+            ray.init(
+                num_cpus=int(num_cpus),
+                include_webui=False,
+                ignore_reinit_error=True,
+                plasma_directory=plasma_directory,
+                object_store_memory=object_store_memory,
+                address=redis_address,
+                redis_password=redis_password,
+                logging_level=100,
+                memory=object_store_memory,
+                lru_evict=True,
+            )
+        else:
+            raise ValueError("cluster variable should be 'True' or None")
```

Comment:
Minor rewording on error for clarity:
```suggestion
            raise ValueError("'MODIN_RAY_CLUSTER' env variable not correctly set! Did you mean `os.environ["MODIN_RAY_CLUSTER"] = "True"`?")
```